metagraph/core/planning.py

```diff
@@ -13,25 +13,33 @@ def __init__(self, src_type):
         self.src_type = src_type
         self.translators = []
         self.dst_types = []
+        self.unsatisfiable = False
 
     def __len__(self):
+        if self.unsatisfiable:
+            raise ValueError(
+                "No translation path found for {src_type.__name__} -> {self.dst_types[-1].__name__}"
```

Comment:
This will break because `self.dst_types` is empty. When you decide that a translation is unsatisfiable, you need to add the final dst_type just so the messages explain where the user wanted to get to.

Reply:
Alternatively, you could add a `.final_type` attribute to the constructor so that we always know the start and desired end types, regardless of what other translation steps or unsatisfiability we tack on later.

Reply:
https://github.com/metagraph-dev/metagraph/pull/110/commits/c964882abd5d7416483bbffba147771b43a85402 addresses this problem by storing the failed destination type in `unsatisfiable_dst_type`, which would act as a flag indicating if the translation failed (`unsatisfiable_dst_type` is not `None`) or has not failed yet (`unsatisfiable_dst_type` is `None`). @jim22k , does this solution sound okay for you?

Reply:
This is a functional solution, so I'm okay with it.

I would have structured it differently, adding a `.final_type` attribute to the constructor and keeping the boolean `unsatisfiable` flag. The reason is that the `unsatisfiable_dst_type` will always be the same as the `.final_type`. It's never an intermediate step that's missing a translator. And with this error flag being public-facing, I find a simple boolean to be a nicer interface than checking whether an error object is populated. But that's just my preference.

Reply:
That sounds like a good idea. 

https://github.com/metagraph-dev/metagraph/pull/110/commits/d987e9fe86152b7eb0a8594c59b1b5ac612ec5be intends to implement that approach. We now have a `final_type` attribute and an `unsatisfiable` attribute. This commit makes it so that the constructor for `MultieStepTranslator` additionally takes a `final_type` parameter.